accord-core/src/main/java/accord/messages/Accept.java

```diff
@@ -39,51 +39,55 @@
 import javax.annotation.Nullable;
 
 import static accord.local.Command.AcceptOutcome.*;
-import static accord.messages.PreAccept.calculatePartialDeps;
 
 // TODO (low priority, efficiency): use different objects for send and receive, so can be more efficient
 //                                  (e.g. serialize without slicing, and without unnecessary fields)
 public class Accept extends TxnRequest.WithUnsynced<Accept.AcceptReply>
 {
     public static class SerializerSupport
     {
-        public static Accept create(TxnId txnId, PartialRoute<?> scope, long waitForEpoch, long minEpoch, boolean doNotComputeProgressKey, Ballot ballot, Timestamp executeAt, Seekables<?, ?> keys, PartialDeps partialDeps, Txn.Kind kind)
+        public static Accept create(TxnId txnId, PartialRoute<?> scope, long waitForEpoch, long minEpoch, boolean doNotComputeProgressKey, Ballot ballot, Timestamp executeAt, Seekables<?, ?> keys, PartialDeps partialDeps)
         {
-            return new Accept(txnId, scope, waitForEpoch, minEpoch, doNotComputeProgressKey, ballot, executeAt, keys, partialDeps, kind);
+            return new Accept(txnId, scope, waitForEpoch, minEpoch, doNotComputeProgressKey, ballot, executeAt, keys, partialDeps);
         }
     }
 
     public final Ballot ballot;
     public final Timestamp executeAt;
     public final Seekables<?, ?> keys;
     public final PartialDeps partialDeps;
-    public final Txn.Kind kind;
 
-    public Accept(Id to, Topologies topologies, Ballot ballot, TxnId txnId, FullRoute<?> route, Timestamp executeAt, Seekables<?, ?> keys, Deps deps, Txn.Kind kind)
+    public Accept(Id to, Topologies topologies, Ballot ballot, TxnId txnId, FullRoute<?> route, Timestamp executeAt, Seekables<?, ?> keys, Deps deps)
     {
         super(to, topologies, txnId, route);
         this.ballot = ballot;
         this.executeAt = executeAt;
         this.keys = keys.slice(scope.covering());
         this.partialDeps = deps.slice(scope.covering());
-        this.kind = kind;
     }
 
-    private Accept(TxnId txnId, PartialRoute<?> scope, long waitForEpoch, long minEpoch, boolean doNotComputeProgressKey, Ballot ballot, Timestamp executeAt, Seekables<?, ?> keys, PartialDeps partialDeps, Txn.Kind kind)
+    private Accept(TxnId txnId, PartialRoute<?> scope, long waitForEpoch, long minEpoch, boolean doNotComputeProgressKey, Ballot ballot, Timestamp executeAt, Seekables<?, ?> keys, PartialDeps partialDeps)
     {
         super(txnId, scope, waitForEpoch, minEpoch, doNotComputeProgressKey);
         this.ballot = ballot;
         this.executeAt = executeAt;
         this.keys = keys;
         this.partialDeps = partialDeps;
-        this.kind = kind;
     }
 
     @Override
     public synchronized AcceptReply apply(SafeCommandStore safeStore)
     {
+        if (minEpoch < txnId.epoch())
```

Comment:
During range movements, we need to adopt dependencies on transactions from earlier epochs. This is part of what we discussed before about how we have diverged from the paper in a way I am not keen on, but is what we have for now: in the paper new replicas do not participate in decisions until they have sufficient information to respond. However, today we instead seek "votes" from every epoch that has not fully migrated to the new replicas, from whom we only need the dependencies they know of.

Reply:
This needs a brief comment, explaining it's to avoid wasting time tracking Accept state when the votes isn't needed only the dependency information since we only replicate this in an earlier unsynchronized epoch.